In [1]:
import pandas as pd

import numpy as np

In [2]:
data=pd.read_csv("appstore_games (2).csv")

In [3]:
data

URL          ID  \
0       https://apps.apple.com/us/app/sudoku/id284921427   284921427   
1      https://apps.apple.com/us/app/reversi/id284926400   284926400   
2      https://apps.apple.com/us/app/morocco/id284946595   284946595   
3      https://apps.apple.com/us/app/sudoku-free/id28...   285755462   
4      https://apps.apple.com/us/app/senet-deluxe/id2...   285831220   
5      https://apps.apple.com/us/app/sudoku-classic-n...   286210009   
6      https://apps.apple.com/us/app/gravitation/id28...   286313771   
7       https://apps.apple.com/us/app/colony/id286363959   286363959   
8        https://apps.apple.com/us/app/carte/id286566987   286566987   
9      https://apps.apple.com/us/app/barrels-o-fun/id...   286682679   
10     https://apps.apple.com/us/app/quaddraxx/id2875...   287563734   
11     https://apps.apple.com/us/app/lumen-lite/id288...   288096268   
12     https://apps.apple.com/us/app/bubblepop/id2886...   288669794   
13      https://apps.apple.com/us/app/marple/id288689440   288689440   
14     https://apps.apple.com/us/app/tetravex-lite/id...   288918962   
15     https://apps.apple.com/us/app/awele-oware-manc...   289217958   
16     https://apps.apple.com/us/app/awele-oware-manc...   289217958   
17     https://apps.apple.com/us/app/chess-game/id289...   289382458   
18     https://apps.apple.com/us/app/catcha-mouse/id2...   290493286   
19     https://apps.apple.com/us/app/cool-sudoku-jigs...   290614255   
20     https://apps.apple.com/us/app/mind-the-corners...   291074572   
21     https://apps.apple.com/us/app/all-you-can-eat/...   291723451   
22     https://apps.apple.com/us/app/ane-rouge/id2918...   291894818   
23     https://apps.apple.com/us/app/neiscat/id291929815   291929815   
24     https://apps.apple.com/us/app/boomshine/id2922...   292242592   
25     https://apps.apple.com/us/app/gaia-lite/id2922...   292274288   
26     https://apps.apple.com/us/app/fieldrunners/id2...   292421271   
27     https://apps.apple.com/us/app/lux-touch-3-worl...   292538570   
28     https://apps.apple.com/us/app/dont-square/id29...   292975610   
29     https://apps.apple.com/us/app/smart-rummy/id29...   294166538   
...                                                  ...         ...   
16977  https://apps.apple.com/us/app/biz-the-corporat...  1474039516   
16978  https://apps.apple.com/us/app/magic-puzzle-box...  1474104832   
16979  https://apps.apple.com/us/app/color-match-puzz...  1474107521   
16980  https://apps.apple.com/us/app/block-soldier-sn...  1474206079   
16981  https://apps.apple.com/us/app/alien-technology...  1474220103   
16982  https://apps.apple.com/us/app/gen-e-ric-space-...  1474259867   
16983  https://apps.apple.com/us/app/magicpresentsibe...  1474297479   
16984  https://apps.apple.com/us/app/window-cleaner/i...  1474340647   
16985  https://apps.apple.com/us/app/bing-bang-go-tic...  1474344095   
16986  https://apps.apple.com/us/app/boys-adventure/i...  1474349448   
16987  https://apps.apple.com/us/app/absurd-monsters-...  1474349467   
16988  https://apps.apple.com/us/app/save-the-soviet/...  1474360857   
16989  https://apps.apple.com/us/app/splash-cube/id14...  1474403250   
16990  https://apps.apple.com/us/app/defend-your-pizz...  1474417405   
16991  https://apps.apple.com/us/app/tetris/id1474442643  1474442643   
16992  https://apps.apple.com/us/app/slip-io/id147446...  1474461240   
16993  https://apps.apple.com/us/app/agadmator-chess-...  1474461379   
16994  https://apps.apple.com/us/app/fourthplayer/id1...  1474479471   
16995  https://apps.apple.com/us/app/justin-turdeau/i...  1474503771   
16996  https://apps.apple.com/us/app/defend-the-glass...  1474541937   
16997  https://apps.apple.com/us/app/stars-merged/id1...  1474600038   
16998  https://apps.apple.com/us/app/acrobatic-animal...  1474601590   
16999  https://apps.apple.com/us/app/number-puzzle-20...  1474605652   
17000  https://apps.apple.com/us/app/super-kid/id1474...  1474608803   
17001  https://apps.a

In [4]:
data.describe(include="all")

URL            ID  \
count                                               17007  1.700700e+04   
unique                                              16847           NaN   
top     https://apps.apple.com/us/app/wild-beyond/id12...           NaN   
freq                                                    2           NaN   
mean                                                  NaN  1.059614e+09   
std                                                   NaN  2.999676e+08   
min                                                   NaN  2.849214e+08   
25%                                                   NaN  8.996543e+08   
50%                                                   NaN  1.112286e+09   
75%                                                   NaN  1.286983e+09   
max                                                   NaN  1.475077e+09   

                         Name        Subtitle  \
count                   17007            5261   
unique                  16847            5010   
top     Zombie Gunner Defense  Emoji Stickers   
freq                        2              14   
mean                      NaN             NaN   
std                       NaN             NaN   
min                       NaN             NaN   
25%                       NaN             NaN   
50%                       NaN             NaN   
75%                       NaN             NaN   
max                       NaN             NaN   

                                                 Icon URL  \
count                                               17007   
unique                                              16847   
top     https://is3-ssl.mzstatic.com/image/thumb/Purpl...   
freq                                                    2   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

        Average User Rating  User Rating Count         Price In-app Purchases  \
count           7561.000000       7.561000e+03  16983.000000             7683   
unique                  NaN                NaN           NaN             3803   
top                     NaN                NaN           NaN             0.99   
freq                    NaN                NaN           NaN              943   
mean               4.060905       3.306531e+03      0.813419              NaN   
std                0.751428       4.232256e+04      7.835732              NaN   
min                1.000000       5.000000e+00      0.000000              NaN   
25%                3.500000       1.200000e+01      0.000000              NaN   
50%                4.500000       4.600000e+01      0.000000              NaN   
75%                4.500000       3.090000e+02      0.000000              NaN   
max                5.000000       3.032734e+06    179.990000              NaN   

       Description                                   Developer Age Rating  \
count        17007                                       17007      17007   
unique       16473                                        8693          4   
top         #NAME?  Tapps Tecnologia da Informa\xe7\xe3o Ltda.         4+   
freq            17                                         123      11806   
mean           NaN                                         NaN        NaN   
std            NaN                                         NaN        NaN   
min            NaN                                         NaN        NaN   
25%            NaN                                         NaN        NaN   
50%            NaN                                         NaN        NaN   
75%            NaN                                         NaN        NaN   
max        

As we can see that columns ***ID, URL, Name, Icon URL and Subtitle*** having all distinct values and do not effect the rating of the apploication. So we can drop them.

In [5]:
data = data.drop("URL", axis=1)
data = data.drop("Name", axis=1)
data = data.drop("Subtitle", axis=1)
data = data.drop("Icon URL", axis=1)
data = data.drop("ID", axis=1)

In [6]:
data.describe(include="all")

Average User Rating  User Rating Count         Price In-app Purchases  \
count           7561.000000       7.561000e+03  16983.000000             7683   
unique                  NaN                NaN           NaN             3803   
top                     NaN                NaN           NaN             0.99   
freq                    NaN                NaN           NaN              943   
mean               4.060905       3.306531e+03      0.813419              NaN   
std                0.751428       4.232256e+04      7.835732              NaN   
min                1.000000       5.000000e+00      0.000000              NaN   
25%                3.500000       1.200000e+01      0.000000              NaN   
50%                4.500000       4.600000e+01      0.000000              NaN   
75%                4.500000       3.090000e+02      0.000000              NaN   
max                5.000000       3.032734e+06    179.990000              NaN   

       Description                                   Developer Age Rating  \
count        17007                                       17007      17007   
unique       16473                                        8693          4   
top         #NAME?  Tapps Tecnologia da Informa\xe7\xe3o Ltda.         4+   
freq            17                                         123      11806   
mean           NaN                                         NaN        NaN   
std            NaN                                         NaN        NaN   
min            NaN                                         NaN        NaN   
25%            NaN                                         NaN        NaN   
50%            NaN                                         NaN        NaN   
75%            NaN                                         NaN        NaN   
max            NaN                                         NaN        NaN   

       Languages          Size Primary Genre                   Genres  \
count      16947  1.700600e+04         17007                    17007   
unique       990           NaN            21                     1004   
top           EN           NaN         Games  Games, Strategy, Puzzle   
freq       12467           NaN         16286                      778   
mean         NaN  1.157064e+08           NaN                      NaN   
std          NaN  2.036477e+08           NaN                      NaN   
min          NaN  5.132800e+04           NaN                      NaN   
25%          NaN  2.295014e+07           NaN                      NaN   
50%          NaN  5.676895e+07           NaN                      NaN   
75%          NaN  1.330271e+08           NaN                      NaN   
max          NaN  4.005591e+09           NaN                      NaN   

       Original Release Date Current Version Release Date  
count                  17007                        17007  
unique                  3084                         2512  
top                2/09/2016                    1/08/2019  
freq                      75                          118  
mean                     NaN                          NaN  
std                      NaN                          NaN  
min                      NaN                          NaN  
25%                      NaN                          NaN  
50%                      NaN                          NaN  
75%                      NaN                          NaN  
max                      NaN                          NaN

Also we don't need ***Description, Developer, Age Rating and any Release Date*** to predict the rating of any product.

In [7]:
data=data.drop("Developer", axis=1)
data=data.drop("Original Release Date", axis=1)
data=data.drop("Current Version Release Date", axis=1)
data=data.drop("Description",axis=1)
data=data.drop("Age Rating", axis=1)

In [8]:
data

Average User Rating  User Rating Count  Price  \
0                      4.0             3553.0   2.99   
1                      3.5              284.0   1.99   
2                      3.0             8376.0   0.00   
3                      3.5           190394.0   0.00   
4                      3.5               28.0   2.99   
5                      3.0               47.0   0.00   
6                      2.5               35.0   0.00   
7                      2.5              125.0   0.99   
8                      2.5               44.0   0.00   
9                      2.5              184.0   0.00   
10                     NaN                NaN   0.00   
11                     3.5             5072.0   0.00   
12                     3.0              526.0   0.00   
13                     3.5              989.0   0.99   
14                     2.5             2358.0   0.00   
15                     3.0              112.0   0.00   
16                     3.0              112.0   0.00   
17                     3.0              504.0   0.00   
18                     3.5            95971.0   0.00   
19                     3.5              357.0   0.00   
20                     3.0                7.0   0.99   
21                     2.0               30.0   0.99   
22                     NaN                NaN   0.00   
23                     NaN                NaN   0.99   
24                     4.0              637.0   0.99   
25                     3.5             1563.0   0.00   
26                     4.0            41869.0   2.99   
27                     3.5            72428.0   0.00   
28                     2.5              190.0   0.99   
29                     2.5               69.0   2.99   
...                    ...                ...    ...   
16977                  NaN                NaN   0.00   
16978                  NaN                NaN    NaN   
16979                  NaN                NaN   0.00   
16980                  NaN                NaN    NaN   
16981                  NaN                NaN   0.00   
16982                  NaN                NaN   0.00   
16983                  NaN                NaN   0.00   
16984                  NaN                NaN   0.00   
16985                  NaN                NaN   0.00   
16986                  NaN                NaN   0.00   
16987                  NaN                NaN   0.99   
16988                  NaN                NaN   0.00   
16989                  NaN                NaN   0.00   
16990                  NaN                NaN   0.00   
16991                  NaN                NaN   0.00   
16992                  NaN                NaN   0.00   
16993                  5.0               33.0   0.00   
16994                  NaN                NaN   0.99   
16995                  NaN                NaN   1.99   
16996                  NaN                NaN   0.00   
16997                  NaN                NaN   0.00   
16998                  NaN                NaN   0.00   
16999                  NaN                NaN   0.00   
17000                  NaN                NaN    NaN   
17001                  NaN                NaN    NaN   
17002                  NaN                NaN   0.00   
17003                  NaN                NaN   0.00   
17004                  NaN                NaN   0.00   
17005                  NaN                NaN   0.00   
17006                  NaN                NaN   0.00   

                               In-app Purchases  \
0                                           NaN   
1                                           NaN   
2                                           NaN   
3                                           NaN   
4                                           NaN   
5                                          1.99   
6                                           NaN   
7                                           NaN   
8                                           NaN   
9                       

In [9]:
len([i for i in list(data["Average User Rating"].isna()) if i==False])

7561

In [10]:
len([i for i in data["Primary Genre"] if i=="Games"])

16286

### Droping the rows with missing User Rating

In [11]:
data=data.dropna(subset=["Average User Rating"])

In [12]:
data.shape


(7561, 8)

In [13]:
len([ i for i in data["Average User Rating"] if i>=4.0])

5573

In [14]:
len([i for i in zip(data["Average User Rating"], data["Primary Genre"]) if i[0]>=4.0 and i[1]=="Games"])

5387

In [15]:
data["Primary Genre"].value_counts()

Games                7291
Entertainment          92
Education              46
Utilities              44
Sports                 23
Reference              18
Finance                 8
Productivity            8
Book                    5
Lifestyle               4
Stickers                3
Music                   3
Social Networking       3
Health & Fitness        3
Business                2
News                    2
Medical                 2
Travel                  1
Navigation              1
Food & Drink            1
Shopping                1
Name: Primary Genre, dtype: int64

In this dataset ***7291*** users out of total ***7561*** users are using applications of Primary Genre ***Games***. And there are ***5573*** ratings which are greater than and equal to 4 out of which ***5387*** ratings are of Primary Genre ***Games***.

In [16]:
data.describe(include="all")

Average User Rating  User Rating Count        Price In-app Purchases  \
count           7561.000000       7.561000e+03  7561.000000             4634   
unique                  NaN                NaN          NaN             2965   
top                     NaN                NaN          NaN             0.99   
freq                    NaN                NaN          NaN              391   
mean               4.060905       3.306531e+03     0.571305              NaN   
std                0.751428       4.232256e+04     2.415658              NaN   
min                1.000000       5.000000e+00     0.000000              NaN   
25%                3.500000       1.200000e+01     0.000000              NaN   
50%                4.500000       4.600000e+01     0.000000              NaN   
75%                4.500000       3.090000e+02     0.000000              NaN   
max                5.000000       3.032734e+06   139.990000              NaN   

       Languages          Size Primary Genre                       Genres  
count       7537  7.561000e+03          7561                         7561  
unique       789           NaN            21                          623  
top           EN           NaN         Games  Games, Simulation, Strategy  
freq        5007           NaN          7291                          289  
mean         NaN  1.514679e+08           NaN                          NaN  
std          NaN  2.550380e+08           NaN                          NaN  
min          NaN  2.158400e+05           NaN                          NaN  
25%          NaN  3.061862e+07           NaN                          NaN  
50%          NaN  7.964672e+07           NaN                          NaN  
75%          NaN  1.776138e+08           NaN                          NaN  
max          NaN  4.005591e+09           NaN                          NaN

In [20]:
import matplotlib.pyplot as plt
import seaborn as sns

ImportError: DLL load failed: The specified procedure could not be found.

In [ ]:
genre_counts = dict(data['Primary Genre'].value_counts())

In [ ]:
distributed_genres={}
for i in genre_counts.keys():
    distributed_genres[i]=data[data['Primary Genre']==i]

In [ ]:
distributed_genres.keys()

In [ ]:
sns.distplot(distributed_genres['Games']['Average User Rating'], kde=False)

In [ ]:
for i in distributed_genres.keys():
    sns.distplot(distributed_genres[i]['Average User Rating'], kde=False)
    plt.title(i)
    plt.show()

#### In the above Histograms we can see that the value count of Games genre is the highest in Primary Genre feature and it has most of the ratings between 4.0 and 5.0 so ***Games*** is the most significant genre in this dataset.

In [ ]:
data["Genres"].value_counts()

The most significant genres are ***Games, Simulation and Strategy*** beacause there are ***289 + 289 = 578*** users are using apps of these genres which is much greater than the second highest Genre combination in the dataset.

In [ ]:
data["Average User Rating"].value_counts()

In [ ]:
a = [i[1] for i in list(zip(data["Average User Rating"], data["Genres"])) if i[0]==5.0]

In [ ]:
a = pd.Series(a)

In [ ]:
a.value_counts()

Apllications of Genres ***Games, Strategy and Puzzle*** are highest rated apps.
There are total ***60 + 64 = 124*** applications of Genres ***Games, Strategy and Puzzle*** having 5.0 rating.

In [ ]:
sns.relplot(x="Average User Rating", y="Price",kind="line", data=data)

#### According to the relational graph between Rating and Price, apps of price approx. ***0.5*** are having nice user ratings.

In [ ]:
sns.relplot(x="Average User Rating", y="Price", data=distributed_genre)